### DISCUSSION TITLE
how can i only get the json body of a post request

### DISCUSSION THREAD

**Discussion by yuxiazff** [fastapi/fastapi#6424] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import uvicorn

app = FastAPI()


@app.post("/test")
def test():
    # ====how to get the json of this post api====
    return "yes"


uvicorn.run(app, host="0.0.0.0", port=8010)
```

### Description

i have a post api， which receive json body, but the json's key is variable, so i cannot use BaseModel to define the key.  i only want to get the json, but i did not find the method. how can i only get the json body of a post request.

### Operating System

Windows

### Operating System Details

windows 10 

### FastAPI Version

0.68.1

### Python Version

3.7.11

**Comment by RomanRahimov** [comment]
@app.post("/test")
async def test(body: Request):
  r = await body.json()
  ...  use r

**Comment by AlexandreBiguet** [comment]
Hi :) 

If you don't control the system that POSTs to this endpoint, it might be worth mentioning that if you have no body at all, or a body that is not json, calling `request_instance.json()` would throw a `json.JSONDecodeError` exception ([here](https://github.com/encode/starlette/blob/master/starlette/requests.py#L235)). 

I get why you might want to handle various representations of a single resource, and it may help if you could require (or kindly ask ?) your user to specify some headers that you would use to determine to what representation you need to cast the body (by defining custom `content-type`s ?).

Cheers

**Comment by yuxiazff** [comment]
@RomanRahimov your solution has solved my problem , thank you!  And thanks @AlexandreBiguet  for your kind !
